src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/store/custom.py

```diff
@@ -0,0 +1,257 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import codecs
+import os
+import re
+import time
+import uuid
+
+from OpenSSL import crypto
+
+from msrestazure.azure_exceptions import CloudError
+from azure.mgmt.datalake.store.models import (DataLakeStoreAccountUpdateParameters,
+                                        FirewallRule,
+                                        DataLakeStoreAccount,
+                                        EncryptionConfigType,
+                                        EncryptionIdentity,
+                                        EncryptionConfig,
+                                        EncryptionState,
+                                        KeyVaultMetaInfo)
+
+from azure.datalake.store.multithread import (ADLUploader, ADLDownloader)
+from azure.cli.command_modules.datalake.store._client_factory import (cf_datalake_store_filesystem)
+
+import azure.cli.core.telemetry as telemetry
+from azure.cli.core._util import CLIError
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+# account customiaztions
+def list_adls_account(client, resource_group_name=None):
+    account_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+    return list(account_list)
+
+def create_adls_account(client, 
+                        resource_group_name,
+                        account_name,
+                        location,
```

Comment:
There's a "validate_location" validator in the network modules _validator.py file that you can copy into your module to accomplish this.